ninja-core/src/main/java/ninja/Router.java

```diff
@@ -79,6 +85,9 @@
      *          or simply use a String. If the raw uri does not contain the placeholders
      *          they will be added as query parameters ?key=value&key2=value2 and so on
      * @return The final url (without server, and without any prefixes)
+     * @deprecated Reverse routing in the Router is not validated and does not
+     *      URL-escape path or query parameters. Use <code>ninja.ReverseRouter</code>
+     *      to build your reverse routes.
      */
```

Comment:
Not sure how i forgot the annotation.  Fixed.  We have a LOT of these in our code and it would be difficult to upgrade everything -- backwards compat here is important.  We can remove later in Ninja 7.